connector/src/yang/connector/netconf.py

```diff
@@ -30,6 +32,88 @@ class BaseConnection:
 # The 'Sending' messages are logged at level INFO.
 # The 'Received' messages are logged at level DEBUG.
 
+LOG_FORMAT = '%(asctime)s: %%NETCONF-%(levelname)s: %(message)s'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+
+def format_xml(msg):
+    parser = et.XMLParser(recover=True, remove_blank_text=True)
+
+    if isinstance(msg, str):
+        msg = msg.encode("utf-8")
+
+    msg = msg.strip()
+
+    start = msg.find(b"<")
+    end = msg.rfind(b"]]>]]>")   # NETCONF 1.0 terminator
+
+    if end == -1:
+        end = msg.rfind(b">")
+        if end != -1:
+            # Include the '>' character in our range
+            end += 1
+
+    if start != -1 and end != -1:
+        try:
+            elem = et.fromstring(msg[start:end], parser)
+            text = et.tostring(elem, pretty_print=True,
+                               encoding="utf-8")
+            msg = (msg[:start] + text + msg[end:])
+        except Exception as err:
+            logger.exception(err)
+
+    return msg.decode()
+
+class NetconfFormatter(logging.Formatter):
+    """
+        For formatting NETCONF XML messages
+    """
+
+    def __init__(self, fmt=LOG_FORMAT, date_fmt=DATE_FORMAT):
+        super().__init__(fmt=fmt,
+                         datefmt=date_fmt)
+
+    def format(self, record):
+        msg = record.msg
+        if isinstance(msg, operations.rpc.RPCReply):
+            msg = msg.xml
+        record.msg = format_xml(msg)
+        return super().format(record)
+
+
+class NetconfScreenFormatter(NetconfFormatter):
+    """
+        For limiting the output for formatted messages to max 40 lines
+    """
+    MAX_LINES = 40
+
+    def format(self, record):
+        msg = super().format(record)
+        lines = msg.splitlines()
+        msg_len = len(lines)
+        if msg_len > self.MAX_LINES:
+            return '\n'.join(lines[0:20] + [f'\n\n... skipping {msg_len-40} lines ...\n\n'] + lines[-20:])
+        else:
+            return '\n'.join(lines)
```

Comment:
Why split NETCONF messages?  The rpc-reply could be large and users may want to copy/paste it from log.

Reply:
The response can be quite large, this is only done for console / tasklog messages. The full response is availble in the per device/connection log file (which is part of this change)

Reply:
Not really clear about what a ``console`` or ``tasklog`` message is.  What I do know about running pyats tests is that I have log messages show up in my terminal when running tests, and, I also have the same log messages showing up in the etrade link in my web browser.  I have used both of those sources to copy/paste content into my VS editor to examine or reuse messages.  With this change, I would have to find and open the log file.  Were there a multiple requests from users to shorten these messages?  If not, can we leave them as-is?

Reply:
Will add some more enhancements for the user control of log messages before merging this

Reply:
> Not really clear about what a `console` or `tasklog` message is. What I do know about running pyats tests is that I have log messages show up in my terminal when running tests, and, I also have the same log messages showing up in the etrade link in my web browser. I have used both of those sources to copy/paste content into my VS editor to examine or reuse messages. With this change, I would have to find and open the log file. Were there a multiple requests from users to shorten these messages? If not, can we leave them as-is?

``MAX_LINES`` is still hard-coded to 40 for RPC reporting.  If you want to keep it this way we can do so and modify later if users complain.